parachain/runtime/tests/test_redeem.rs

```diff
@@ -642,14 +642,13 @@ fn integration_test_redeem_wrapped_cancel_liquidated_reimburse() {
         // NOTE: changes are relative the the post liquidation state
         assert_eq!(
             ParachainState::get(),
-            post_liquidation_state.with_changes(|user, vault, liquidation_vault, fee_pool, _| {
+            post_liquidation_state.with_changes(|user, vault, liquidation_vault, _, _| {
                 // to-be-redeemed decreased, forwarding to liquidation vault
                 vault.to_be_redeemed -= redeem.amount_btc + redeem.transfer_fee_btc;
                 liquidation_vault.to_be_redeemed -= redeem.amount_btc + redeem.transfer_fee_btc;
 
                 // tokens are given to the vault, minus a fee that is given to the fee pool
                 vault.free_tokens += redeem.amount_btc + redeem.transfer_fee_btc;
-                fee_pool.vault_wrapped_rewards += vault_rewards(redeem.fee);
```

Comment:
Why isn't the fee_pool given a share anymore? What happens to this fee? (also not that the comment above now is incorrect)

Reply:
It is but in this case when the SLA is taken to zero, all rewards are withdrawn since there is only one vault in this test.

Reply:
But only the stake is withdrawn, right? I mean, any earned rewards are not withdrawn? I think withdrawing stake should not change the total amount of money in the fee pool.

Reply:
Ah yes, so actually once the vault is liquidated it is now removed from the reward pool. Once `cancel_redeem` is called all rewards are distributed to the maintenance fund since `total_stake` is now zero.

Reply:
Can you point me to where everything is distributed to the maintainance fund? Also, to be clear, liquidation does not mean the vault loses all rewards it has earned so far, right?

Reply:
https://github.com/interlay/btc-parachain/blob/master/crates/fee/src/lib.rs#L298

Reply:
> Also, to be clear, liquidation does not mean the vault loses all rewards it has earned so far, right?

Correct

Reply:
> https://github.com/interlay/btc-parachain/blob/master/crates/fee/src/lib.rs#L298

Could we change the distribute function like this?
```
fn distribute(reward: Self::SignedFixedPoint) -> Result<Self::SignedFixedPoint, DispatchError> {
        let total_stake = Self::total_stake();
        // if total_stake.is_zero() {
        //     return Ok(Self::SignedFixedPoint::zero());
        // }

        let reward_div_total_stake = reward
            .checked_div(&total_stake)
            .or(0);
        checked_add_mut!(RewardPerToken<T, I>, &reward_div_total_stake);
        checked_add_mut!(TotalRewards<T, I>, &reward);
        Ok(reward)
    }
    ```
    
I feel like this results in fewer edge cases that we have to test. Right now, you are removing an important test. Either we need to add another test to explicitly check what happens if there _is_ a vault remaining in the system, or we make the change I proposed

Reply:
@sander2 since investigating distribution without stake is not the purpose of this PR shall we merge and ticket? Or shall I refactor the tests to allow for an additional participant who would receive all rewards upon liquidation?